plugin/storage/badger/spanstore/reader.go

```diff
@@ -612,3 +619,45 @@ func scanRangeFunction(it *badger.Iterator, indexEndValue []byte) bool {
 	}
 	return false
 }
+
+// loadServices fills the cache with services after extracting from badger
+func (r *TraceReader) loadServices() {
```

Comment:
```suggestion
func (r *TraceReader) preloadServices() {
```
if it's called only once